Ambit/Source/Ambit/Mode/ConfigImportExport.spec.cpp

```diff
@@ -1017,4 +1017,133 @@ void ConfigImportExportSpec::Define()
+
+            TestTrue("Reply should be true on return", Reply.IsEventHandled());
+        });
+
+        AfterEach([this]()
+        {
+            GLevelEditorModeTools().DeactivateMode(FAmbitMode::EM_AmbitModeId);
+            Exporter = nullptr;
+            JsonContent = "";
+        });
+    });
+
+    Describe("OnExportGltf()", [this]()
+    {
+        BeforeEach([this]()
+        {
+            // Generate World
+            World = FAutomationEditorCommonUtils::CreateNewMap();
+
+            // Create Mode Settings
+            GLevelEditorModeTools().ActivateMode(FAmbitMode::EM_AmbitModeId);
+
+            FAmbitMode* AmbitMode = FAmbitMode::GetEditorMode();
+            check(AmbitMode);
+
+            // Generate Exporter
+            Exporter = NewObject<UMockableConfigImportExport>();
+
+            auto MockListBuckets = []() -> TSet<FString>
+            {
+                TSet<FString> setBuckets;
+                setBuckets.Add("BucketName");
+                return setBuckets;
+            };
+            Exporter->SetMockS3ListBuckets(MockListBuckets);
+
+            auto MockCreateBucket = [](const FString& Region, const FString& BucketName) -> void
+            {
+                return;
+            };
+            Exporter->SetMockS3CreateBucket(MockCreateBucket);
+
+            auto MockS3FileUpload = [](const FString& Region, const FString& BucketName, const FString& ObjectName, const FString& FilePath) -> bool
+            {
+                return true;
+            };
+            Exporter->SetMockS3FileUpload(MockS3FileUpload);
+        });
+
+        It("Should fail if no mesh to export", [this]()
+        {
+            AddExpectedError("Cannot find a static mesh to export.", EAutomationExpectedErrorFlags::Exact, 1);
+            Exporter->OnExportGltf();
+        });
+
+        It("Should succeed after test map is loaded", [this]()
+        {
+            // Load the test map.
+            // Loading this map resets some aspects of AmbitMode and the Exporter objects. These need to be reset again
+            // in order for the test to succeed.
+            FAutomationEditorCommonUtils::LoadMap("/Ambit/Test/Maps/ProceduralPlacementTestMap");
+
+            // Create Mode Settings
+            GLevelEditorModeTools().ActivateMode(FAmbitMode::EM_AmbitModeId);
+
+            FAmbitMode* AmbitMode = FAmbitMode::GetEditorMode();
+            check(AmbitMode);
+
+            // This function has been set in BeforeEach() but loading a new map causes it to be unset. Most likely has
+            // to do with state management in UE. Resetting this function should stop any exceptions from being thrown.
+            auto MockS3FileUpload = [](const FString& Region, const FString& BucketName, const FString& ObjectName, const FString& FilePath) -> bool
+            {
+                return true;
+            };
+            Exporter->SetMockS3FileUpload(MockS3FileUpload);
```

Comment:
I consistently get an exception on this line unless I re-instantiate `Exporter` to a new object. Something has caused it to become corrupted due to ` FAutomationEditorCommonUtils::LoadMap("/Ambit/Test/Maps/ProceduralPlacementTestMap");` being called. Either needs refactored or you can remove the load map for your own object additions.

Reply:
removed `LoadMap()` and have instead added a single object to the scene for export instead.